src/scripts/heroAnimation/hero.js

```diff
@@ -0,0 +1,256 @@
+}
+
+function mouseEnter() {
+  initAnimation();
+}
+
+function mouseOut() {
+  target.x = container.clientHeight * 2;
+  target.y = container.clientHeight * 2;
+  setTimer();
+}
+
+function getDistance(p1, p2) {
+  return Math.pow(p1.x - p2.x, 2) + Math.pow(p1.y - p2.y, 2);
+}
+
+function addListeners() {
+  let timer = null;
+
+  if (!("ontouchstart" in window)) {
+    container.addEventListener("mousemove", mouseMove);
+    container.addEventListener("mouseout", mouseOut);
+    container.addEventListener("mouseenter", mouseEnter);
+    window.addEventListener("resize", resizeCanvas);
+  }
+
+  window.addEventListener("scroll", scrollCheck);
+
+  window.onresize = function() {
+    location.reload();
+  };
+  setTimer();
+}
+
+function initHeader() {
+  resizeCanvas();
+  width = container.clientWidth;
+  height = container.clientHeight;
+  canvas.width = width;
+  canvas.height = height;
+  target = {
+    x: width * 2,
+    y: height * 2
+  };
+
+  if (typeof canvas.getContext !== "function") {
+    console.info("the thingy bob doesn't support canvas. Bailing out.");
+    return;
+  }
+
+  ctx = canvas.getContext("2d");
+  target = {
+    x: width * 2,
+    y: height * 2
+  };
+
+  // create points - added breakpoint with less nodes displayed as on smaller screens as it can look very dense
+  if (window.matchMedia("screen and (min-width:768px)").matches) {
+    points = [];
+    for (var x = 0; x < width; x = x + width / 9) {
+      for (var y = 0; y < height; y = y + height / 9) {
+        var px = x + Math.random() * width / 9;
+        var py = y + Math.random() * height / 9;
+        var p = {
+          x: px,
+          originX: px,
+          y: py,
+          originY: py
+        };
+        points.push(p);
+      }
+    }
+  } else {
+    points = [];
+    for (var x = 0; x < width; x = x + width / 6) {
+      for (var y = 0; y < height; y = y + height / 6) {
+        var px = x + Math.random() * width / 6;
+        var py = y + Math.random() * height / 6;
+        var p = {
+          x: px,
+          originX: px,
+          y: py,
+          originY: py
+        };
+        points.push(p);
+      }
+    }
+  }
+
+  // for each point find the 5 closest points
+  for (var i = 0; i < points.length; i++) {
+    var closest = [];
+    var p1 = points[i];
+    for (var j = 0; j < points.length; j++) {
+      var p2 = points[j];
+      if (!(p1 == p2)) {
+        var placed = false;
+        for (var k = 0; k < 5; k++) {
+          if (!placed) {
+            if (closest[k] == undefined) {
+              closest[k] = p2;
+              placed = true;
+            }
+          }
+        }
+
+        for (var k = 0; k < 5; k++) {
+          if (!placed) {
+            if (getDistance(p1, p2) < getDistance(p1, closest[k])) {
+              closest[k] = p2;
+              placed = true;
+            }
+          }
+        }
+      }
+    }
+    p1.closest = closest;
+  }
+
+  // assign a circle to each point
+  for (var i in points) {
```

Comment:
I would also totally suggest to improve readability of this code with some functional programming kata ;) (e.g. `array.prototype.map()`).

Reply:
This will be refactored in another story